fastly/erl.go

```diff
@@ -0,0 +1,342 @@
+	ERLLogLoggly          ERLLoggerEnum = "loggly"
+	ERLLogLogShuttle      ERLLoggerEnum = "logshuttle"
+	ERLLogNewRelic        ERLLoggerEnum = "newrelic"
+	ERLLogOpenStack       ERLLoggerEnum = "openstack"
+	ERLLogPaperTrail      ERLLoggerEnum = "papertrail"
+	ERLLogPubSub          ERLLoggerEnum = "pubsub"
+	ERLLogS3              ERLLoggerEnum = "s3"
+	ERLLogScalyr          ERLLoggerEnum = "scalyr"
+	ERLLogSftp            ERLLoggerEnum = "sftp"
+	ERLLogSplunk          ERLLoggerEnum = "splunk"
+	ERLLogStackDriver     ERLLoggerEnum = "stackdriver"
+	ERLLogSumoLogiuc      ERLLoggerEnum = "sumologic"
+	ERLLogSysLog          ERLLoggerEnum = "syslog"
+)
+
+// Enum for ERL
+type ERLWindowSizeEnum int
+
+const (
+	ERLSize1  ERLWindowSizeEnum = 1
+	ERLSize10 ERLWindowSizeEnum = 10
+	ERLSize60 ERLWindowSizeEnum = 60
+)
+
+// ERLsByName is a sortable list of ERLs
+type ERLsByName []*ERL
+
+// Len, Swap, and Less implement the sortable interface
+func (s ERLsByName) Len() int      { return len(s) }
+func (s ERLsByName) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s ERLsByName) Less(i, j int) bool {
+	return s[i].Name < s[j].Name
+}
+
+// ********************************************************************************************* //
+//
+// Function 1: List rate limiters
+// Note(s):
+//
+// Return type
+//   []*ERL
+
+// Input to the ListERLs function
+type ListERLsInput struct {
+	ServiceID      string // required
+	ServiceVersion int    // required
+}
+
+// Returns the list of ERLs for the specified version
+func (c *Client) ListERLs(i *ListERLsInput) ([]*ERL, error) {
+	if i.ServiceID == "" {
+		return nil, ErrMissingServiceID
+	}
+	if i.ServiceVersion == 0 {
+		return nil, ErrMissingServiceVersion
+	}
+
+	path := fmt.Sprintf("/service/%s/version/%d/rate-limiters", i.ServiceID, i.ServiceVersion)
+	resp, err := c.Get(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var erls []*ERL
+	if err := decodeBodyMap(resp.Body, &erls); err != nil {
+		return nil, err
+	}
+
+	sort.Stable(ERLsByName(erls))
+	return erls, nil
+}
+
+// ********************************************************************************************* //
+//
+// Function 2: Create rate limiter
+// Note(s):
+//   VersionId must not be locked
+//
+// Return type
+//   *ERL
+
+// Input to the CreateERL function
+type CreateERLInput struct {
+	ServiceID          string               `json:"-"`
+	ServiceVersion     int                  `json:"-"`
+	Action             ERLActionEnum        `json:"action"`
+	ClientKey          []string             `json:"client_key"`
+	HttpMethods        []ERLHttpMethodsEnum `json:"http_methods"`
+	Name               string               `json:"name"`
+	PenaltyBoxDuration int                  `json:"penalty_box_duration"`
+	Response           *ERLResponseType     `json:"response,omitempty"`
+	RpsLimit           int                  `json:"rps_limit"`
+	WindowSize         ERLWindowSizeEnum    `json:"window_size"`
```

Comment:
Sorted and `Enum` suffix removed...

```suggestion
	Action             ERLAction        `json:"action"`
	ClientKey          []string             `json:"client_key"`
	HttpMethods        []ERLHttpMethods `json:"http_methods"`
	Name               string               `json:"name"`
	PenaltyBoxDuration int                  `json:"penalty_box_duration"`
	Response           *ERLResponseType     `json:"response,omitempty"`
	RpsLimit           int                  `json:"rps_limit"`
	ServiceID          string               `json:"-"`
	ServiceVersion     int                  `json:"-"`
	WindowSize         ERLWindowSize    `json:"window_size"`
```